torch/csrc/jit/codegen/cuda/lower_misaligned_vectorization.cpp

```diff
@@ -516,13 +516,6 @@ class MisalignedVectorizationModifier : public kir::ExprMutator {
     auto namedScalar = (address) ? IrBuilder::addressExprNamedScalar(name, val)
                                  : IrBuilder::setExprNamedScalar(name, val);
     TORCH_INTERNAL_ASSERT(namedScalar->definition() != nullptr);
-
-    auto alloc = IrBuilder::create<kir::Allocate>(
-        namedScalar,
-        MemoryType::Local,
-        GpuLower::current()->kernel()->oneVal());
-    body.push_back(alloc);
-    body.push_back(namedScalar->definition());
     return namedScalar;
```

Comment:
It is fine now mostly because my change in `codegen.cpp` treats named scalar just like any other scalar. Here is the code generated for `NVFuserTest.FusionVectorizeMisalignedPointwise_CUDA`:

```CUDA
__global__ void kernel1(Tensor<float, 2> T0, Tensor<float, 2> T1, Tensor<float, 2> T2) {
  NVFUSER_DEFINE_MAGIC_ZERO
  int i43;
  i43 = ((nvfuser_index_t)blockIdx.x) * T0.size[1];
  int i38;
  i38 = ((nvfuser_index_t)threadIdx.x) * 4;
  #pragma unroll 1
  for(nvfuser_index_t i27 = 0; i27 < (ceilDiv(T0.size[1], 256)); ++i27) {
    int i114;
    i114 = i27 * 256;
    Array<float, 4, 4> T4;
    Array<float, 4, 4> T3;
    // Alias Allocation - register
    auto& T5 = T3;
    if (((((i27 * 256) + i38) + (where(((4 - (((int64_t) &T1[i43 + ((i27 * 256) + i38)] % (4 * 4)) / 4)) == 4), 0, (4 - (((int64_t) &T1[i43 + ((i27 * 256) + i38)] % (4 * 4)) / 4))))) < (T0.size[1] - ((T0.size[1] - (where(((4 - (((int64_t) &T1[i43 + ((i27 * 256) + i38)] % (4 * 4)) / 4)) == 4), 0, (4 - (((int64_t) &T1[i43 + ((i27 * 256) + i38)] % (4 * 4)) / 4))))) % 4)))) {
      loadGlobalToLocal<float, 4, false>(&T4[0],  &T1[i43 + (((i27 * 256) + i38) + (where(((4 - (((int64_t) &T1[i43 + ((i27 * 256) + i38)] % (4 * 4)) / 4)) == 4), 0, (4 - (((int64_t) &T1[i43 + ((i27 * 256) + i38)] % (4 * 4)) / 4)))))]);
      loadGlobalToLocal<float, 4, false>(&T3[0],  &T0[i43 + (((i27 * 256) + i38) + (where(((4 - (((int64_t) &T1[i43 + ((i27 * 256) + i38)] % (4 * 4)) / 4)) == 4), 0, (4 - (((int64_t) &T1[i43 + ((i27 * 256) + i38)] % (4 * 4)) / 4)))))]);
      #pragma unroll
      for(nvfuser_index_t i26 = 0; i26 < 4; ++i26) {
        T5[i26]
          = T3[i26]
          + T4[i26];
      }
      NVFUSER_UPDATE_MAGIC_ZERO
      loadLocalToGlobal<float, 4, false>( &T2[(((nvfuser_index_t)blockIdx.x) * T0.size[1]) + ((i114 + i38) + (where(((4 - (((int64_t) &T1[i43 + ((i27 * 256) + i38)] % (4 * 4)) / 4)) == 4), 0, (4 - (((int64_t) &T1[i43 + ((i27 * 256) + i38)] % (4 * 4)) / 4)))))], &T5[0]);
    }
    if ((((i27 * 256) + i38) == 0)) {
      #pragma unroll
      for(nvfuser_index_t i25 = 0; i25 < 4; ++i25) {
        if ((i25 < (where(((4 - (((int64_t) &T1[i43 + ((i27 * 256) + i38)] % (4 * 4)) / 4)) == 4), 0, (4 - (((int64_t) &T1[i43 + ((i27 * 256) + i38)] % (4 * 4)) / 4)))))) {
          T4[i25]
             = T1[i43 + (i114 + (i38 + (i25 + nvfuser_zero)))];
        }
      }
      NVFUSER_UPDATE_MAGIC_ZERO
      #pragma unroll
      for(nvfuser_index_t i22 = 0; i22 < 4; ++i22) {
        if ((i22 < (where(((4 - (((int64_t) &T1[i43 + ((i27 * 256) + i38)] % (4 * 4)) / 4)) == 4), 0, (4 - (((int64_t) &T1[i43 + ((i27 * 256) + i38)] % (4 * 4)) / 4)))))) {
          T3[i22]
             = T0[i43 + (i114 + (i38 + (i22 + nvfuser_zero)))];
        }
      }
      NVFUSER_UPDATE_MAGIC_ZERO
      #pragma unroll
      for(nvfuser_index_t i26 = 0; i26 < 4; ++i26) {
        if ((i26 < (where(((4 - (((int64_t) &T1[i43 + ((i27 * 256) + i38)] % (4 * 4)) / 4)) == 4), 0, (4 - (((int64_t) &T1[i43 + ((i27 * 256) + i38)] % (4 * 4)) / 4)))))) {
          T5[i26]
            = T3[i26]
            + T4[i26];
        }
      }
      NVFUSER_UPDATE_MAGIC_ZERO
      #pragma unroll
      for(nvfuser_index_t i28 = 0; i28 < 4; ++i28) {
        if ((i28 < (where(((4 - (((int64_t) &T1[i43 + ((i27 * 256) + i38)] % (4 * 4)) / 4)) == 4), 0, (4 - (((int64_t) &T1[i43 + ((i27 * 256) + i38)] % (4 * 4)) / 4)))))) {
          T2[(((nvfuser_index_t)blockIdx.x) * T0.size[1]) + (i114 + (i38 + (i28 + nvfuser_zero)))]
             = T5[i28];
        }
      }
      NVFUSER_UPDATE_MAGIC_ZERO
    }
    if ((((((i27 * 256) + i38) + (where(((4 - (((int64_t) &T1[i43 + ((i27 * 256) + i38)] % (4 * 4)) / 4)) == 4), 0, (4 - (((int64_t) &T1[i43 + ((i27 * 256) + i38)] % (4 * 4)) / 4))))) >= (T0.size[1] - ((T0.size[1] - (where(((4 - (((int64_t) &T1[i43 + ((i27 * 256) + i38)] % (4 * 4)) / 4)) == 4), 0, (4 - (((int64_t) &T1[i43 + ((i27 * 256) + i38)] % (4 * 4)) / 4))))) % 4))) && ((((i27 * 256) + i38) + (where(((4 - (((int64_t) &T1[i43 + ((i27 * 256) + i38)] % (4 * 4)) / 4)) == 4), 0, (4 - (((int64_t) &T1[i43 + ((i27 * 256) + i38)] % (4 * 4)) / 4))))) < T0.size[1]))) {
      #pragma unroll
      for(nvfuser_index_t i25 = 0; i25 < 4; ++i25) {
        if ((i25 < ((T0.size[1] - (where(((4 - (((int64_t) &T1[i43 + ((i27 * 256) + i38)] % (4 * 4)) / 4)) == 4), 0, (4 - (((int64_t) &T1[i43 + ((i27 * 256) + i38)] % (4 * 4)) / 4))))) % 4))) {
          T4[i25]
             = T1[i43 + ((i114 + (i38 + (i25 + nvfuser_zero))) + (where(((4 - (((int64_t) &T1[i43 + ((i27 * 256) + i38)] % (4 * 4)) / 4)) == 4), 0, (4 - (((int64_t) &T1[i43 + ((i27 * 256) + i38)] % (4 * 4)) / 4)))))];
        }
      }
      NVFUSER_UPDATE_MAGIC_ZERO
      #pragma unroll
      for(nvfuser_index_t i22 = 0; i22 < 4; ++i22) {
        if ((i22 < ((T0.size[1] - (where(((4 - (((int64_t) &T1[i43 + ((i27 * 256) + i38)] % (4 * 4)) / 4)) == 4), 0, (4 - (((int64_t) &T1[i43 + ((i27 * 256) + i38)] % (4 * 4)) / 4))))) % 4))) {
          T3[i22]
             = T0[i43 + ((i114 + (i38 + (i22 + nvfuser_zero))) + (where(((4 - (((int64_t) &T1[i43 + ((i27 * 256) + i38)] % (4 * 4)) / 4)) == 4), 0, (4 - (((int64_t) &T1[i43 + ((i27 * 256) + i38)] % (4 * 4)) / 4)))))];
        }
      }
      NVFUSER_UPDATE_MAGIC_ZERO
      #pragma unroll
      for(nvfuser_index_t i26 = 0; i26 < 4; ++i26) {
        if ((i26 < ((T0.size[1] - (where(((4 - (((int64_t) &T1[i43 + ((i27 * 256) + i38)] % (4 * 4)) / 4)) == 4), 0, (4 - (((int64_t) &T1[i43 + ((i27 * 256) + i38)] % (4 * 4)) / 4))))) % 4))) {
          T5[i26]
            = T3[i26]
            + T4[i26];
        }
      }
      NVFUSER_UPDATE_MAGIC_ZERO
      #pragma unroll
      for(nvfuser_index_t i28 = 0; i28 < 4; ++i28) {
        if ((i28 < ((T0.size[1] - (where(((4 - (((int64_t) &T1[i43 + ((i27 * 256) + i38)] % (4 * 4)) / 4)) == 4), 0, (4 - (((int64_t) &T1[i43 + ((i27 * 256) + i38)] % (4 * 4)) / 4))))) % 4))) {
          T2[(((nvfuser_index_t)blockIdx.x) * T0.size[1]) + ((i114 + (i38 + (i28 + nvfuser_zero))) + (where(((4 - (((int64_t) &T1[i43 + ((i27 * 256) + i38)] % (4 * 4)) / 4)) == 4), 0, (4 - (((int64_t) &T1[i43 + ((i27 * 256) + i38)] % (4 * 4)) / 4)))))]
             = T5[i28];
        }
      }
      NVFUSER_UPDATE_MAGIC_ZERO
    }
  }
}
```